packages/components/src/notice/types.ts

```diff
@@ -0,0 +1,109 @@
+/**
+ * External dependencies
+ */
+import type { MouseEventHandler, ReactNode } from 'react';
+
+type CommonNoticeActionProps = {
+	label: string;
+	className?: string;
+	noDefaultClasses?: boolean;
+	variant?: 'primary' | 'secondary' | 'link';
+};
+// `url` and `onClick` can both be provided, but `url` takes precedence. If
+// `url` is provided, the action's button will be rendered as an anchor and
+// `onClick` will be ignored.
+type NoticeActionWithURL = CommonNoticeActionProps & {
+	url: string;
+	onClick?: never;
+};
+type NoticeActionWithOnClick = CommonNoticeActionProps & {
+	url?: never;
+	onClick: MouseEventHandler< HTMLButtonElement >;
+};
+
+export type NoticeAction = NoticeActionWithURL | NoticeActionWithOnClick;
+
+export type NoticeChildren = string | JSX.Element;
+
+export type NoticeProps = {
+	/**
+	 * A CSS `class` to give to the wrapper element.
+	 */
+	className?: string;
+	/**
+	 * The displayed message of a notice. Also used as the spoken message for
+	 * assistive technology, unless `spokenMessage` is provided as an alternative message.
+	 */
+	children: ReactNode;
+	/**
+	 * Used to provide a custom spoken message in place of the `children` default.
+	 *
+	 * @default `children`
+	 */
+	spokenMessage?: ReactNode;
+	/**
+	 * Can be warning (yellow), success (green), error (red), or info.
+	 *
+	 * @default 'info'
+	 */
+	status?: 'warning' | 'success' | 'error' | 'info';
+	/**
+	 * Function called when dismissing the notice
+	 *
+	 * @default noop
+	 */
+	onRemove?: () => void;
+	/**
+	 * A politeness level for the notice's spoken message. Should be provided as
+	 * one of the valid options for an aria-live attribute value. If not provided,
+	 * a sensible default is used based on the notice status. Note that this
+	 * value should be considered a suggestion; assistive technologies may
+	 * override it based on internal heuristics.
+	 */
+	politeness?: 'polite' | 'assertive';
```

Comment:
The previous description for this prop in `Snackbar` was quite more verbose — I wonder if we should keep the more verbose version around? Also, we should sync the JSDoc descriptions in this file with the README

Reply:
Addressed both of these (more verbose `politeness` doc and syncing up the README in aa43e4efe8